businessCentral/src/ADLSEExternalEvents.Codeunit.al

```diff
@@ -2,16 +2,54 @@
 // Licensed under the MIT License. See LICENSE in the project root for license information.
 codeunit 82574 "ADLSE External Events"
 {
-    internal procedure OnTableExportRunEnded(RunId: Integer; TableId: Integer; State: Enum "ADLSE Run State")
+    Access = Internal;
+    SingleInstance = true;
+
+    var
+        Environment, Resource : Text[250];
+
+    procedure OnTableExportRunEnded(RunId: Integer; Started: DateTime; Ended: DateTime; TableId: Integer; State: Enum "ADLSE Run State")
     var
-        ADLSESetup: Record "ADLSE Setup";
         ADLSEUtil: Codeunit "ADLSE Util";
     begin
+        SetEnvironmentAndResource();
+
+        ExportEntityEnded(RunId, Started, Ended, State, Environment, Resource, CopyStr(ADLSEUtil.GetDataLakeCompliantTableName(TableId), 1, 250));
+#pragma warning disable AL0432
+        TableExportRunEnded(RunId, State, Environment, CopyStr(ADLSEUtil.GetDataLakeCompliantTableName(TableId), 1, 250));
+#pragma warning restore AL0432
+    end;
+
+    procedure OnExportRunEnded(Ended: DateTime)
+    begin
+        SetEnvironmentAndResource();
+        ExportEnded(Ended, Environment, Resource);
+    end;
+
+    local procedure SetEnvironmentAndResource()
+    var
+        ADLSESetup: Record "ADLSE Setup";
+    begin
+        if (Environment <> '') and (Resource <> '') then
+            exit;
+
         ADLSESetup.GetSingleton();
-        TableExportRunEnded(RunId, State, ADLSESetup.Container, CopyStr(ADLSEUtil.GetDataLakeCompliantTableName(TableId), 1, 250));
+        Environment := ADLSESetup."Account Name";
+        Resource := ADLSESetup.Container;
+    end;
+
+    [ExternalBusinessEvent('ExportEnded', 'Export ended', 'The export of all the entities was registered as ended.', EventCategory::ADLSE)]
+    local procedure ExportEnded(Ended: DateTime; Environment: Text[250]; Resource: Text[250])
+    begin
+    end;
+
+    [ExternalBusinessEvent('ExportEntityEnded', 'Export entity ended', 'The export of the entity was registered as ended.', EventCategory::ADLSE)]
+    local procedure ExportEntityEnded(RunId: Integer; Started: DateTime; Ended: DateTime; State: Enum "ADLSE Run State"; Environment: Text[250]; Resource: Text[250]; Entity: Text[250])
```

Comment:
Maybe Area a better name? 
Enviroment is indeed double iwth Power Automate.

Reply:
Good feedback! I've thought again on the naming and would like to go for Instance. I've ralso emoved the ExportEntityEnded event for now and indeed better to create a separate pull-request for this.